cmd/ebrelayer/contract/Peggy.sol

```diff
@@ -0,0 +1,733 @@
+        emit Approval(msg.sender, spender, _allowed[msg.sender][spender]);
+        return true;
+    }
+
+    /**
+    * @dev Transfer token for a specified addresses
+    * @param from The address to transfer from.
+    * @param to The address to transfer to.
+    * @param value The amount to be transferred.
+    */
+    function _transfer(address from, address to, uint256 value) internal {
+        require(to != address(0));
+
+        _balances[from] = _balances[from].sub(value);
+        _balances[to] = _balances[to].add(value);
+        emit Transfer(from, to, value);
+    }
+
+    /**
+     * @dev Internal function that mints an amount of the token and assigns it to
+     * an account. This encapsulates the modification of balances such that the
+     * proper events are emitted.
+     * @param account The account that will receive the created tokens.
+     * @param value The amount that will be created.
+     */
+    function _mint(address account, uint256 value) internal {
+        require(account != address(0));
+
+        _totalSupply = _totalSupply.add(value);
+        _balances[account] = _balances[account].add(value);
+        emit Transfer(address(0), account, value);
+    }
+
+    /**
+     * @dev Internal function that burns an amount of the token of a given
+     * account.
+     * @param account The account whose tokens will be burnt.
+     * @param value The amount that will be burnt.
+     */
+    function _burn(address account, uint256 value) internal {
+        require(account != address(0));
+
+        _totalSupply = _totalSupply.sub(value);
+        _balances[account] = _balances[account].sub(value);
+        emit Transfer(account, address(0), value);
+    }
+
+    /**
+     * @dev Internal function that burns an amount of the token of a given
+     * account, deducting from the sender's allowance for said account. Uses the
+     * internal burn function.
+     * Emits an Approval event (reflecting the reduced allowance).
+     * @param account The account whose tokens will be burnt.
+     * @param value The amount that will be burnt.
+     */
+    function _burnFrom(address account, uint256 value) internal {
+        _allowed[account][msg.sender] = _allowed[account][msg.sender].sub(value);
+        _burn(account, value);
+        emit Approval(account, msg.sender, _allowed[account][msg.sender]);
+    }
+}
+
+  /*
+   *  @title: Processor
+   *  @dev: Processes requests for item locking and unlocking by
+   *        storing an item's information then relaying the funds
+   *        the original sender.
+   */
+contract Processor {
+
+    using SafeMath for uint256;
+
+    /*
+    * @dev: Item struct to store information.
+    */    
+    struct Item {
+        address payable sender;
+        bytes recipient;
+        address token;
+        uint256 amount;
+        uint256 nonce;
+        bool locked;
+    }
+
+    uint256 public nonce;
+    mapping(bytes32 => Item) private items;
```

Comment:
@denali

Reply:
Each Item's id is a hash containing: sender's contract address on the Ethereum network, specified recipient address on the Bridge network, the token type's deployed address on the Ethereum network, the amount, and the current nonce. The hashed id provides the same uniqueness while intrinsically providing additional context about the Item's contents.  

There's been some discussion around having validators sign a message containing this hash as opposed to an EthBridgeClaim which contains each individual piece of information; but as this is not reflected in the current implementation, I think simplifying the unique id to just the nonce is a reasonable suggestion. The Item could still contain a bytes32 hash of the transaction details, allowing any future iteration of the project to leverage it.  

@musnit what are your thoughts on this change? It will impact the contracts/test suite significantly, as well as the Relayer's LockEvent struct.

Reply:
@denalimarsh I think it depends on if the added complexity of using a hash on-chain actually brings additional security/verifiability to the process in any way, either now in futures. As it is, I don't see that is - how do you see it possibly doing so in future? Unless there is a clear potential future benefit, I'd also suggest simplifying it.

Reply:
I've created an issue for this https://github.com/cosmos/peggy/issues/55